gcovr/utils.py

```diff
@@ -179,6 +179,11 @@ def build_filter(self, logger):
             logger.warn("did you mean: {}", suggestion)
 
         if os.path.isabs(self.regex):
+            if use_canonical_paths:
+                canonical_path = os.path.realpath(self.regex)
+                if canonical_path != self.regex:
+                    logger.msg(f"Filter {self.regex} has been normalized to {canonical_path}.")
+                    self.regex = canonical_path
```

Comment:
This code has to be removed. The regex cannot be treated as a path. The design of the filter system distinguishes between different filter types so that in the match() function, the path being matched is transformed to the correct form. So we process the path to fit the regex, not the regex to fit the path. This was a major design improvement that fixed *many* bugs.

So instead, the `AbsoluteFilter()` might need a `use_canonical_path` property. However, it already seems correct. See a few lines below:

```python
class AbsoluteFilter(Filter):
    def match(self, path):
        abspath = os.path.realpath(path)
        return super(AbsoluteFilter, self).match(abspath)
```

This approach is not perfect. Especially for a RelativeFilter involving symlinks/drive mappings, this could lead to scenarios where a match becomes impossible. But it's far less buggy behaviour overall.

Could you take a look at the Windows code in RelativeFilter? I think it is also correct for your purposes, but please check. We resolve the realpath() of the path and the project root, and directly fail the match if the drive letters don't match.

Reply:
Looking at the filter code, I agree that the current one seems to be correct already. Will remove this code.